lib/js/jsbigints.nim

```diff
@@ -0,0 +1,133 @@
+## Arbitrary precision integers.
+## * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
+when not defined(js) and not defined(nimdoc):
+  {.fatal: "Module jsbigints is designed to be used with the JavaScript backend.".}
+
+type JsBigInt* = ref object of JsRoot ## Arbitrary precision integer for JavaScript target.
+
+func newBigInt*(integer: SomeInteger): JsBigInt {.importjs: "BigInt(#)".}
+  ## Constructor for `JsBigInt`.
+
+func big*(integer: cstring): JsBigInt {.importjs: "BigInt(#)".}
+  ## Constructor for `JsBigInt`.
+
+func toLocaleString*(this: JsBigInt): cstring {.importjs: "#.$1()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toLocaleString*(this: JsBigInt; locales: cstring): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toLocaleString*(this: JsBigInt; locales: openArray[cstring]): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toLocaleString
+
+func toString*(this: JsBigInt; radix: int): cstring {.importjs: "#.$1(#)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func toString*(this: JsBigInt): cstring {.importjs: "#.toString()".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/toString
+
+func `$`*(this: JsBigInt): string = $toString(this)
+  ## Return a string representation of `JsBigInt`.
+
+func asIntN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asIntN
+
+func asUintN*(width: int; bigInteger: JsBigInt): int {.importjs: "BigInt.$1(#, #)".}
+  ## https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/asUintN
+
+func `+`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `-`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `*`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)".}
+
+func `div`*(x, y: JsBigInt): JsBigInt {.importjs: "(# / #)".}
+   ## Same as `div` but for `JsBigInt`.
+
+func `mod`*(x, y: JsBigInt): JsBigInt {.importjs: "(# % #)".}
+  ## Same as `mod` but for `JsBigInt` (uses JavaScript `BigInt() % BigInt()`).
+
+func `+=`*(x, y: JsBigInt): JsBigInt {.importjs: "(# $1 #)", discardable.}
```

Comment:
change to:
```
func `+=`*(x, y: JsBigInt) {.importjs: "(# $1 #)".}
```
likewise for every proc you marked as discardable.

* discardable should generally be avoided and is error prone, it causes surprises in generic code (eg with auto-return) that ends in a discardable.
* `+=` in rest of nim returns void
* you can get the same effect via the explicit: `let b = (a+=c; a)` so there's no loss of functionality in the rare case where you need this